api/server/server.go

```diff
@@ -1123,6 +1126,142 @@ func postContainersCopy(eng *engine.Engine, version version.Version, w http.Resp
 	return nil
 }
 
+// handleArchiveError is a convenience function for the following Copy
+// handlers which makes a JSON error message but leaves writing the
+// header and response to the package's common error handler.
+func handleArchiveError(w http.ResponseWriter, err error) error {
```

Comment:
I would really rather not touch the existing `httpError()` function in this PR. I wrote this handler to nicely format any errors as JSON objects in the response body **and** that could still be handled correctly be the existing `httpError()` function.